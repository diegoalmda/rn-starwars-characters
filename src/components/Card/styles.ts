import { StyleSheet } from 'react-native';
import { colors } from '../../globalStyles';

export const styles = StyleSheet.create({
  container: {
    flex: 1,
    width: '100%',
    borderColor: colors.gray,
    borderWidth: 1,
    marginBottom: 10,
    borderRadius: 10,
    padding: 10,
    backgroundColor: colors.purple_light,
  },
  textName: {
    fontSize: 25,
    color: colors.gray,
    marginBottom: 10,
  },
  itemText: {
    color: colors.gray,
  },
  detailsContainer: {
    flex: 1,
    flexDirection: 'row',
    justifyContent:'space-between',
    alignItems: 'center',
    marginBottom: 10,
  }
})