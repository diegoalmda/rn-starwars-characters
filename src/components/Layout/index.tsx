import { FC, ReactNode } from 'react';
import { StatusBar } from 'expo-status-bar';
import { ImageBackground, Platform, SafeAreaView, StyleSheet } from 'react-native';

import backgroundImage from '../../assets/background.png';

interface LayoutProps {
  children: ReactNode
}

export const Layout:FC = ({ children }: LayoutProps) => {
  return (
    <ImageBackground 
      style={styles.container}
      source={backgroundImage}
      defaultSource={backgroundImage}
      imageStyle={styles.cover}
    >
      <StatusBar
        style='light'
        backgroundColor='transparent'
        translucent
      />
      <SafeAreaView style={styles.AndroidSafeArea}>
        { children }
      </SafeAreaView>
    </ImageBackground>
  )
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#000',
    alignItems: 'center',
    justifyContent: 'center',
  },
  cover: {
    width: '100%',
    height: '100%',
    position: 'absolute',
    resizeMode: 'cover',
    opacity: 0.5,
  },
  AndroidSafeArea: {
    flex: 1, 
    marginVertical: Platform.OS === 'android' ? 30 : 0,
    width: '100%',
    paddingHorizontal: 10,
  }
});