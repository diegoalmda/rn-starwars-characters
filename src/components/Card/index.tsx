import { FC } from "react";
import { View, Text } from "react-native";
import { styles } from './styles';

interface CardProps {
  characterInfo: {
    name: string;
    height: string;
    mass: string;
  }
}

export const Card:FC<CardProps> = ({ characterInfo }: CardProps) => {
  return (
    <View style={styles.container} >
      <Text style={styles.textName}>{characterInfo.name}</Text>
      <View>
        <Text style={styles.itemText}>Height: {characterInfo.height}</Text>
        <Text style={styles.itemText}>Mass: {characterInfo.mass}</Text>
      </View>
    </View>
  );
}