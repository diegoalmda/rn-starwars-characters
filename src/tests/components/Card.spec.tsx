import React from 'react';
import { Card } from "../../components/Card";
import { cleanup, render } from "@testing-library/react-native";
 
afterEach(cleanup);

describe('Layout Component', () => {
  
  it('renders Child component', () => {
    const characterInfo = {
      name: "John",
      height: "180",
      mass: "100"
    }
 
    const {toJSON, getByText} = render(<Card characterInfo={characterInfo} />); 
    const foundNameText = getByText('John'); 
    expect(foundNameText.props.children).toEqual('John');
    expect(toJSON()).toMatchSnapshot();
  });
});