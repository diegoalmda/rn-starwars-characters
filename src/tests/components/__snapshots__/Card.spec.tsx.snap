// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout Component renders Child component 1`] = `
<View
  style={
    {
      "backgroundColor": "rgba(120, 111, 165, 0.1)",
      "borderColor": "#C3C4C7",
      "borderRadius": 10,
      "borderWidth": 1,
      "flex": 1,
      "marginBottom": 10,
      "padding": 10,
      "width": "100%",
    }
  }
>
  <Text
    style={
      {
        "color": "#C3C4C7",
        "fontSize": 25,
        "marginBottom": 10,
      }
    }
  >
    John
  </Text>
  <View>
    <Text
      style={
        {
          "color": "#C3C4C7",
        }
      }
    >
      Height: 
      180
    </Text>
    <Text
      style={
        {
          "color": "#C3C4C7",
        }
      }
    >
      Mass: 
      100
    </Text>
  </View>
</View>
`;
